### ISSUE TITLE
Step level log 

### ISSUE THREAD

**Issue by xinnjie** [tektoncd/results#535]
### Feature request
Currently logs of all steps in one `Taskrun` is mixed up, we should provide step level log, so that querying the log of one specific step becomes possible.

### Use case
* [Tekton  Dashboard integrations] dashboard demand the function of querying  log of one specific step.
* [Performance enhancement]  logs sometimes are quite large, if one queries log of one specific step, it's no need to query log of other steps.

**Comment by xinnjie** [comment]
One possible implementation:

### Model changes:
[`LogStatus`](https://github.com/tektoncd/results/blob/6bce96faf26f62ce35662a806c9258a2cc8b40c4/pkg/apis/v1alpha2/types.go#L47-L51) 
before:
```go 
// LogStatus defines the current status of the log resource.
type LogStatus struct {
	Path string `json:"path,omitempty"`
	Size int64  `json:"size"`
}
```
after: 
```go 
// LogStatus defines the current status of the log resource.
type LogStatus struct {
	Path string `json:"path,omitempty"` //  Deprecated
	Size int64  `json:"size"`           //  Deprecated

	Steps []StepLogStatus `json:"steps"`
}

// StepLogStatus defines the current status of step log
type StepLogStatus struct {
	Name string `json:"name"` // Step name
	Path string `json:"path"`
	Size int64  `json:"size"`
}
```

**Comment by xinnjie** [comment]
cc @adambkaplan @sayan-biswas

**Comment by sayan-biswas** [comment]
Wouldn't this create a metadata entry for each step in the database? This would be too granular and would require multiple database trip to build the complete log of the taskrun. IMO, if there's a approach which can create one metadata log for a result but the physical files will be stored in the steps level. Then we can provide featured in the API to request only a particular step.

**Comment by xinnjie** [comment]
> Wouldn't this create a metadata entry for each step in the database? 
No,  one log record for one `Taskrun`. Current implementation is intuitive and good enough.


> create one metadata log for a result but the physical files will be stored in the steps level

Yes , I think so too.

One possible implementation would be, one log record for one `Taskrun`(just same as current implementation), and change `LogStatus` like below, every step log store in different physical files.
```go
// LogStatus defines the current status of the log resource.
type LogStatus struct {
	Path string `json:"path,omitempty"` //  Deprecated
	Size int64  `json:"size"`           //  Deprecated

	Steps []StepLogStatus `json:"steps"`
}

// StepLogStatus defines the current status of step log
type StepLogStatus struct {
	Name string `json:"name"` // Step name
	Path string `json:"path"`
	Size int64  `json:"size"`
}
```

**Comment by xinnjie** [comment]
Look like  we all agree on providing feature in the API to request only a particular step?
What left is API/Model changes and implementation detail.

**Comment by adambkaplan** [comment]
Something to consider is that we will want to support delegated log forwarding (say, via fluentd). This was called out as a potential long-term feature in [TEP-0117](https://github.com/tektoncd/community/blob/main/teps/0117-tekton-results-logs.md#implementation-plan). I've filed #619 to make this more concrete.

How steps are represented may depend on the log forwarder deployed, and how logs are represented in the storage backend.

**Comment by cmorinupgrade** [comment]
@adambkaplan was there any progress with regards to this issue? Thank you.

**Comment by sayan-biswas** [comment]
> @adambkaplan was there any progress with regards to this issue? Thank you.

Not yet. :-(
But we are still considering this feature and working on a design and maybe a POC soon.

**Comment by sayan-biswas** [comment]
Before this we are trying to get tekton result use the `v1` tekton APIs and an overall stability to the existing tekton results APIs, and specially the watcher component.
